jenkins-agent.ps1

```diff
@@ -89,6 +90,14 @@ if(![System.String]::IsNullOrWhiteSpace($Cmd)) {
         }
     }
 
+    $AgentArguments = @()
+
+    if(![System.String]::IsNullOrWhiteSpace($JavaOpts)) {
+        $AgentArguments += @($JavaOpts.split(" "))
```

Comment:
What happens if the JavaOpts contains a quoted string with spaces?

Reply:
I guess it doesn't work. I found some scripts like https://github.com/beatcracker/Powershell-Misc/blob/master/Split-CommandLine.ps1 that would be able to do the splitting correctly but I don't have the correct environment to test it atm.

Reply:
I can test out when I am back from my vacation.

Reply:
Is it actually necessary to split it ? maybe it can just be added as is.. 

Per https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/start-process?view=powershell-7.2#parameters:

```
Arguments can be accepted as a single string with the arguments separated by spaces, or as an array of strings separated by commas. The cmdlet joins the array into a single string with each element of the array separated by a single space.

The outer quotes of the PowerShell strings are not included when the ArgumentList values are passed to the new process. If parameters or parameter values contain a space or quotes, they need to be surrounded with escaped double quotes. For more information, see [about_Quoting_Rules](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_quoting_rules?view=powershell-7.2).
```

So we should be able to do something like this no ? 

```
        if (![System.String]::IsNullOrWhiteSpace($javaOpts)) {
            $AgentArguments = @(`"$javaOpts`", $AgentArguments)
        }
```

I haven't tested that but noticed this as part of https://github.com/jenkinsci/docker-inbound-agent/pull/261

Reply:
I think the whole `$AgentArguments` should be a string in that case.

Reply:
possible to write a test for this?

https://github.com/jenkinsci/docker-inbound-agent/blob/master/tests/inboundAgent.Tests.ps1

Reply:
Ah, nice, this wasn't there when I initially wrote this PR. Will have a look.

Reply:
For info: https://github.com/jenkins-infra/helpdesk/issues/3274